packages/aws-cdk/lib/api/aws-auth/credential-plugins.ts

```diff
@@ -44,11 +44,15 @@ export class CredentialPlugins {
 
       // Backwards compatibility: if the plugin returns a ProviderChain, resolve that chain.
       // Otherwise it must have returned credentials.
-      if ((providerOrCreds as any).resolvePromise) {
-        return (providerOrCreds as any).resolvePromise();
-      }
-      return providerOrCreds;
+      const credentials = (providerOrCreds as any).resolvePromise ? await (providerOrCreds as any).resolvePromise() : providerOrCreds;
```

Comment:
minor: could we do better than `as any`? What type are we expecting here?

Reply:
>       // Backwards compatibility: if the plugin returns a ProviderChain, resolve that chain.
>      // Otherwise it must have returned credentials.